### PR TITLE
amazonq: Add Walkthrough page

### PR DESCRIPTION
**Pull request by nkomonen-amazon**
## Problem
We want a walkthrough for Amazon Q. Also some separate changes.

## Other Changes

- Fix Amazon Q debug scripts. A `serve` task was failing constantly before this.
- Enable Dev mode in Amazon Q

### Walkthrough Changes

- Creates the Amazon Q walkthrough
- GIFs still need to be created, we are waiting on those but we can merge for now
- When the user logs in to Amazon Q the first time they will be shown the walkthrough page. It will never show again on subsequent logins
- We have telemetry for when the walkthrough is shown through our "first login" automation, and when certain events are triggered in the walkthrough such as opening the chat

Icon and GIFs will be added later:
Screenshot 2024-04-18 at 5 58 08 PM

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazonq/src/extensionShared.ts [hunk 1]
```diff
@@ -61,6 +62,8 @@ export async function activateShared(context: vscode.ExtensionContext) {
     await activateCWChat(context)
     await activateQGumby(extContext as ExtContext)
 
+    await activateDev(context)
```
Comment (justinmk3) [inline packages/amazonq/src/extensionShared.ts:65]: I think activateDev needs to be owned by Toolkit. won' this create duplicate commands?

File: packages/core/src/amazonq/onboardingPage/walkthrough.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ */
+export async function showAmazonQWalkthroughOnce(
+    state = globals.context.globalState,
+    showWalkthrough = () => openAmazonQWalkthrough.execute()
+) {
+    const hasShownWalkthroughId = 'aws.amazonq.hasShownWalkthrough'
+    const hasShownWalkthrough = state.get(hasShownWalkthroughId, false)
+    if (hasShownWalkthrough) {
+        return
+    }
```
Comment (justinmk3) [inline packages/core/src/amazonq/onboardingPage/walkthrough.ts:23]: Perhaps we need something like `PromptSettings` , but for globalState? Or can we use PromptSettings for this?

**Review by justinmk3 — APPROVED** [review]

File: packages/amazonq/src/extensionShared.ts (hunk 1 above)
Comment (nkomonen-amazon) [reply packages/amazonq/src/extensionShared.ts]: Good catch. activateDev has been removed from this PR, I'll properly implement it in a different one.

File: packages/core/src/amazonq/onboardingPage/walkthrough.ts (hunk 2 above)
Comment (nkomonen-amazon) [reply packages/core/src/amazonq/onboardingPage/walkthrough.ts:23]: Yeah, I think we should create something similar for globalState. Theres probably a lot of duplicate stuff we can remove.

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**